### ISSUE TITLE
Rendered Markdown is not exported appropriately to PDF 

### ISSUE THREAD

**Issue by nombreinvicto** [vsch/idea-multimarkdown#682]
Using:

Pycharm 2018.3.3
OS: Win 10
Plugin: v 2.7 enhanced edition

The markdown document renders mathematical equations fine in the IDE, however, when exported to PDF, it doesnt export the  rendered view but rather exports the code written for the math equations. 

For example, for the following latex code in the document with Katex scripts included:

`$$\Pr\left[\sum_{i=1}^k X_i > c \right] \leq 2^{-\Omega(c^2 k)}$$
$$\sum_{i=1}^\infty \frac{1}{2^i} = 1$$`

The markdown renders as follows:

<https://imgur.com/jvIa3tS>

BUT, when exported to PDF, the PDF appears as follows:

<https://imgur.com/VOHQKcq>

**Comment by vsch** [comment]
@nombreinvicto, PDF conversion uses [OpenHtmlToPDF](https://openhtmltopdf.com) library which cannot make use of JavaScript code which does the Katex and Mermaid math and chart conversion.

It is a limitation of the conversion however, if you use

```markdown
embedded LaTex
![Latex](http://latex.codecogs.com/png.latex?
\Pr\left[\sum_{i=1}^k X_i > c \right] \leq 2^{-\Omega(c^2 k)}$$ $$\sum_{i=1}^\infty
\frac{1}{2^i} = 1
)
in a paragraph.
```

For inline math, to render in preview (but not GitHub):

embedded LaTex ![Latex](http://latex.codecogs.com/png.latex? \Pr\left[\sum_{i=1}^k X_i > c \right] \leq 2^{-\Omega(c^2 k)}$$ $$\sum_{i=1}^\infty \frac{1}{2^i} = 1 ) in a paragraph.

And the same as stand-alone image for block math, then the PDF will render correctly in the PDF.

[682.pdf](https://github.com/vsch/idea-multimarkdown/files/2775862/682.pdf)

To have GitHub display the images correctly, you need to convert the multi-markdown image link to HTML with the intention:

![screenshot_multilineurl_tohtmlintention](https://user-images.githubusercontent.com/10299086/51431027-74abdf80-1bf1-11e9-9902-74734de836f3.png)

Resulting in:

```markdown
embedded LaTex
<img src="http://latex.codecogs.com/png.latex?%5CPr%5Cleft%5B%5Csum_%7Bi=1%7D%5Ek%20X_i%20%3E%20c%20%5Cright%5D%20%5Cleq%202%5E%7B-%5COmega(c%5E2%20k)%7D$$%20$$%5Csum_%7Bi=1%7D%5E%5Cinfty%0A%5Cfrac%7B1%7D%7B2%5Ei%7D%20=%201%0A" alt="Latex" />
in a paragraph.
```

Which renders correctly on all markdown processors.

embedded LaTex <img src="http://latex.codecogs.com/png.latex?%5CPr%5Cleft%5B%5Csum_%7Bi=1%7D%5Ek%20X_i%20%3E%20c%20%5Cright%5D%20%5Cleq%202%5E%7B-%5COmega(c%5E2%20k)%7D$$%20$$%5Csum_%7Bi=1%7D%5E%5Cinfty%0A%5Cfrac%7B1%7D%7B2%5Ei%7D%20=%201%0A" alt="Latex" /> in a paragraph.

To edit the LaTex, use the HTML to Markdown intention on the HTML image tag to get the original markdown:

![screenshot_multilineurl_tomdintention](https://user-images.githubusercontent.com/10299086/51431032-7f667480-1bf1-11e9-8ddd-a4fa31325f16.png)

I will add intentions to convert Katex inline math and block math elements to `latex.codecogs.com` multi-line URL images and back to make the conversion convenient. Similar to the conversion between PlantUML fenced code and Gravizo, in the latest EAP.

**Comment by nombreinvicto** [comment]
Thanks a lot for the feedback! That resolved the issue. Thanks for the great Plugin!

**Comment by vsch** [comment]
@nombreinvicto, option to render Math block and inline elements using codecogs.com was added to HTML Generation settings: [Html Generation Settings: math rendering](https://github.com/vsch/idea-multimarkdown/wiki/Html-Generation-Settings#math-rendering)

Also added intentions to change between Math block/inline and codecogs.com mutli-line URL image links.

The SVG rendering is not handled by the PDF conversion so select the codecogs.com PNG option.

I tried to get JLaTexMath library but form some reason it would not render to SVG nor PNG correctly. Must be some of its dependencies conflicting with versions the IDE loads or some other issue.

I will try other libraries but only after the next version release which is already a week behind schedule.
